### ISSUE TITLE
Spec should clarify what hypervisor/VM interfaces are?

### ISSUE THREAD

**Issue by asridharan** [containernetworking/cni#390]
It is my understanding the CNI spec was specifically for describing interfaces between container run-time and network plugins for linux containers which use network namespaces for isolation. However, in the latest spec I see the following description for interfaces:
>  Hypervisor/VM-based plugins should return an ID unique to the virtualized sandbox the interface was created in. 
This is a bit confusing. Are we now looking at CNI plugins for VM based environments as well? This greatly increases the scope of what CNI was intended for. The maintainers should clarify the scope and give explicit examples of what they mean by these `hypervisor/VM-based plugins`?

**Comment by bboreham** [comment]
@asridharan thanks for the question; we discussed it at the maintainers' meeting today.

There was general agreement that we do want to allow CNI to be used with hypervisor-based container systems.

That said, we also want to avoid adding a lot of extra code in the central repo to support this usage. This repo should be a small core of basic plugins and library code; things beyond that should live in separate repos.

**Comment by asridharan** [comment]
@bboreham understandable. The problem this statement in the SPEC is very confusing is that the SPEC starts by saying the following:
> container can be considered synonymous with a Linux network namespace. What unit this corresponds to depends on a particular container runtime implementation: for example, in implementations of the App Container Spec like rkt, each pod runs in a unique network namespace. In Docker, on the other hand, network namespaces generally exist for each separate Docker container.

The scope above is very limited to containers based on linux network namespaces. If CNI is intended to be used by hypervisor-based container systems (I am assuming this implies VMs), we should be explicit about it. Maybe, I am lacking an understanding of network isolation is achieved (from an implementation standpoint) by something like kvm or Xen.

**Comment by bboreham** [comment]
Agreed, we need to clean up the language so it isn't directly contradictory.

> give explicit examples of what they mean by these `hypervisor/VM-based plugins`?

I can do that: [hyper_](https://hypercontainer.io/), [Intel® Clear Containers](https://clearlinux.org/documentation/clear-containers.html).  These are just examples, not an exclusive list and not implying any recommendation.

**Comment by asridharan** [comment]
Very interesting. Yeah I think we should explicitly call this out if possible.

**Comment by fabiand** [comment]
Please include http://github.com/kubevirt as well. Compared to hyper and clearcontainers, kubevirt uses a different mechanism to spawn VMs, however, those VMs also need network connectivity.

**Comment by bboreham** [comment]
We discussed this at the maintainers' meeting, and felt we don't have enough concrete information to make a spec change.

So, if one or more of the interested parties would like to propose a change, ideally one that they are using in practice with a non-namespace runtime, we'd like to hear it.
